### DISCUSSION TITLE
space-y-* applies to child with position fixed

### DISCUSSION THREAD

**Discussion by kresli** [tailwindlabs/tailwindcss#7452] [Help] [answered]
Is this expected behaviour? I would expect fixed elements to ignore any margin passed from a parent if top-left position is set

```html
<div class="flex flex-col space-y-10">
  <div>content A</div>
  <div>content B</div>
  <div>content C</div>
  <div>content D</div>
  <div class="fixed top-0 left-0 bg-black/20">
    fixed element
  </div>
</div>
```
![image](https://user-images.githubusercontent.com/3826097/153849846-61c18fbc-db05-4eae-ab2a-a548bd47a3df.png)

**Comment by adamwathan** [comment, marked as answer]
Yep this is expected, those classes are just a quick way to apply what is effectively this CSS:

```
.space-y-4 > *:not(:first-child) {
  margin-top: 1rem;
}
```

No real way for us to make it smart enough to account for things like you're describing. Options are:

1. Structure your markup differently so the fixed element isn't a direct child of the `space-y-*` container.
2. Use `gap` with flexbox in the column direction or a single column grid, which will be smart enough to handle this.
3. Use regular margins on each element instead of the `space-y-*` utilities.

Hope that helps.

**Reply by DavidAbraham082** [reply]
Is there a general rule for when it's better to choose `space-*` or `gap-*`?

**Reply by ivan-kleshnin** [reply]
Use `space-*` when you don't want/need to make parent a Flexbox. When it's `flex`, then `gap-*` is the most logical option.

**Comment by ivan-kleshnin** [comment]
You can also add `!mt-0` to the positioned element as a workaround. Just `mt-0` would still be overriden.
